src/net_processing.cpp

```diff
@@ -2605,6 +2609,39 @@ bool SendMessages(CNode* pto, CConnman& connman, std::atomic<bool>& interruptMsg
                 pto->fDisconnect = true;
             }
         }
+        // Check for headers sync timeouts
+        if (state.fSyncStarted && state.nHeadersSyncTimeout < std::numeric_limits<int64_t>::max()) {
+            // Detect whether this is a stalling initial-headers-sync peer
+            if (pindexBestHeader->GetBlockTime() <= GetAdjustedTime() - 24*60*60) {
```

Comment:
I don't know of any other special meaning for 144 blocks besides the one you mentioned tbh. Same applies for 2 weeks (i.e. 2016 blocks, diff adjustment period). But since we already have this logic (to follow similar number of blocks and to keep some logic for timespan even though it doesn't apply directly due to dynamic diff adjustment) in other parts of Dash legacy code, we simply have to follow it in new parts too to be consistent. Or change this all at once if we decide to do this one day, but this kind of changes should definitely be in a separate PR then.